precompile/contract/interfaces.go

```diff
@@ -34,14 +34,11 @@ type StateDB interface {
 	GetState(common.Address, common.Hash) common.Hash
 	SetState(common.Address, common.Hash, common.Hash)
 
-	SetCode(common.Address, []byte)
-
 	SetNonce(common.Address, uint64)
```

Comment:
why did we remove these functions from precompile?

Reply:
They were not used, so we should keep the interfaces narrower when possible.
I believe it was used prior to the refactoring.

Reply:
Ok, makes sense but what if future precompiles want to use these functions? I'm ok either way